backend/src/Repository/ChatRoomEntityRepository.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace App\Repository;
+
+use App\Entity\ChatRoomEntity;
+use App\Entity\StoreOwnerProfileEntity;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Persistence\ManagerRegistry;
+use Doctrine\ORM\Query\Expr\Join;
+use App\Constant\ChatRoom\ChatRoomConstant;
+
+/**
+ * @method ChatRoomEntity|null find($id, $lockMode = null, $lockVersion = null)
+ * @method ChatRoomEntity|null findOneBy(array $criteria, array $orderBy = null)
+ * @method ChatRoomEntity[]    findAll()
+ * @method ChatRoomEntity[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class ChatRoomEntityRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, ChatRoomEntity::class);
+    }
+
+    public function getChatRoomsWithStores(): ?array
+     {   
+        return $this->createQueryBuilder('chatRoom')
+
+            ->select('chatRoom.roomId')
+            ->addSelect('storeOwnerProfile.id as storeOwnerProfileId, storeOwnerProfile.storeOwnerName, storeOwnerProfile.images')
+          
+            ->leftJoin(StoreOwnerProfileEntity::class, 'storeOwnerProfile', Join::WITH, 'chatRoom.userId = storeOwnerProfile.storeOwnerId')
+
+            ->andWhere('chatRoom.usedAs = :usedAs')
+            ->setParameter('usedAs', ChatRoomConstant::ADMIN_STORE)
+
```

Comment:
We may order the result.

Reply:
the target is get chat room id between admin and stores.